lib/gitsh/git_repository.rb

```diff
@@ -62,8 +62,10 @@ def config(name, default = nil, force_default_git_command = false)
       out, err, status = Open3.capture3(command)
       if status.success?
         out.chomp
+      elsif block_given?
+        yield
       else
-        default
+        raise KeyError, "Git configuration variable #{name} is not set"
```

Comment:
I am wondering if these changes make sense in this case - if we are going to raise an error when the config var isn't set, why not just call it normally rather than `fetch`? Is it just so the error message is more readable?